# Classify HFpEF

In [1]:
import pandas as pd
import numpy as np

from preproc import get_hfpef_405, convert2np

In [2]:
lbl_colname= ['Death', 'CV death', 'Major cardiac events']

## Clean Data

In [3]:
data_df = get_hfpef_405()

In [4]:
data_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 405 entries, 0 to 404
Data columns (total 50 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Age                   405 non-null    int64  
 1   Sex                   405 non-null    int64  
 2   Cr                    405 non-null    float64
 3   GFR                   405 non-null    float64
 4   CKD stage             405 non-null    int64  
 5   smoke                 405 non-null    float64
 6   BMI                   405 non-null    float64
 7   BSA                   405 non-null    float64
 8   DM                    405 non-null    int64  
 9   Insulin               405 non-null    int64  
 10  AF                    405 non-null    int64  
 11  SBP                   405 non-null    int64  
 12  DBP                   405 non-null    int64  
 13  MAP                   405 non-null    float64
 14  PP                    405 non-null    int64  
 15  1/2SBP                4

In [5]:
data_df.describe()

,Age,Sex,Cr,GFR,CKD stage,smoke,BMI,BSA,DM,Insulin,...,E/A,Mitral E/e',TR Vmax,RWT,LV mass index,LVH,LAVI,LACI,LA diameter,LVEF
count,405.000000,405.000000,405.000000,405.000000,405.000000,405.000000,405.000000,405.000000,405.000000,405.000000,...,405.000000,405.000000,405.000000,405.000000,405.000000,405.000000,405.000000,405.000000,405.000000,405.000000
mean,72.212346,1.676543,1.554074,54.568272,2.723457,0.069136,26.802571,1.637419,0.575309,0.185185,...,0.725010,16.733481,1.846543,0.585780,113.941951,2.718519,35.029136,3.428062,3.821259,62.316049
std,11.261269,0.468374,1.308252,24.305747,0.991333,0.253999,6.584225,0.230020,0.494907,0.388928,...,0.585953,8.661718,1.410611,0.347155,50.865016,1.996229,26.521582,4.009693,1.342424,8.612943
min,31.000000,1.000000,0.550000,2.000000,1.000000,0.000000,15.400000,1.120000,0.000000,0.000000,...,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
25%,65.000000,1.000000,0.880000,38.000000,2.000000,0.000000,22.400000,1.500000,0.000000,0.000000,...,0.500000,11.200000,0.000000,0.370000,87.200000,1.000000,18.000000,0.000000,3.600000,57.000000
50%,74.000000,2.000000,1.150000,55.000000,3.000000,0.000000,26.200000,1.600000,1.000000,0.000000,...,0.700000,15.900000,2.430000,0.500000,113.000000,3.000000,34.900000,2.710843,4.000000,62.000000
75%,81.000000,2.000000,1.580000,71.000000,3.000000,0.000000,30.500000,1.800000,1.000000,0.000000,...,1.000000,22.100000,2.990000,0.720000,139.600000,4.000000,50.500000,5.563107,4.489300,67.000000
max,97.000000,2.000000,10.000000,108.000000,5.000000,1.000000,55.900000,2.300000,1.000000,1.000000,...,4.100000,42.200000,4.550000,2.300000,356.600000,7.000000,140.800000,23.665480,7.000000,82.900000


## Get Cleaned Data

In [6]:
data_df_nomiss = get_hfpef_405(nomiss=True)

In [8]:
data_df_nomiss.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 145 entries, 1 to 404
Data columns (total 50 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Age                   145 non-null    int64  
 1   Sex                   145 non-null    int64  
 2   Cr                    145 non-null    float64
 3   GFR                   145 non-null    float64
 4   CKD stage             145 non-null    int64  
 5   smoke                 145 non-null    float64
 6   BMI                   145 non-null    float64
 7   BSA                   145 non-null    float64
 8   DM                    145 non-null    int64  
 9   Insulin               145 non-null    int64  
 10  AF                    145 non-null    int64  
 11  SBP                   145 non-null    int64  
 12  DBP                   145 non-null    int64  
 13  MAP                   145 non-null    float64
 14  PP                    145 non-null    int64  
 15  1/2SBP                1

In [9]:
data_df_nomiss.describe()

,Age,Sex,Cr,GFR,CKD stage,smoke,BMI,BSA,DM,Insulin,...,E/A,Mitral E/e',TR Vmax,RWT,LV mass index,LVH,LAVI,LACI,LA diameter,LVEF
count,145.000000,145.000000,145.000000,145.000000,145.000000,145.000000,145.000000,145.000000,145.000000,145.000000,...,145.000000,145.000000,145.000000,145.000000,145.000000,145.000000,145.000000,145.000000,145.000000,145.000000
mean,71.137931,1.662069,1.495862,57.476138,2.586207,0.048276,26.388983,1.622948,0.627586,0.193103,...,0.958621,19.368897,1.935793,0.612117,123.608276,3.082759,41.753103,5.875199,4.039503,62.730345
std,11.297505,0.474644,1.248331,25.388228,1.010955,0.215092,6.303231,0.208832,0.485124,0.396102,...,0.429899,8.612584,1.371418,0.329848,42.068062,2.019019,18.469833,3.981753,0.665267,7.601474
min,31.000000,1.000000,0.600000,2.000000,1.000000,0.000000,15.400000,1.200000,0.000000,0.000000,...,0.400000,4.290000,0.000000,0.230000,53.800000,0.000000,12.500000,1.695652,2.100000,50.000000
25%,64.000000,1.000000,0.870000,39.000000,2.000000,0.000000,21.900000,1.500000,0.000000,0.000000,...,0.700000,12.800000,0.000000,0.400000,94.400000,1.000000,29.300000,3.482659,3.659000,57.300000
50%,73.000000,2.000000,1.160000,60.000000,3.000000,0.000000,25.600000,1.600000,1.000000,0.000000,...,0.900000,17.500000,2.390000,0.522000,117.800000,4.000000,38.900000,4.821151,4.000000,61.000000
75%,79.000000,2.000000,1.520000,75.000000,3.000000,0.000000,29.600000,1.800000,1.000000,0.000000,...,1.200000,23.500000,3.050000,0.700000,145.100000,4.000000,48.300000,6.955836,4.400000,67.000000
max,92.000000,2.000000,10.000000,108.000000,5.000000,1.000000,44.400000,2.300000,1.000000,1.000000,...,3.200000,42.200000,4.550000,2.300000,356.600000,7.000000,107.700000,23.665480,6.500000,80.000000


## Random Forest

In [10]:
from sklearn.model_selection import GridSearchCV
from sklearn.metrics import classification_report

from sklearn.ensemble import RandomForestClassifier
from sklearn.neural_network import MLPClassifier

from sklearn.model_selection import cross_val_score, cross_val_predict, cross_validate 
from sklearn.metrics import classification_report

In [11]:
def hfpef_rf(X, y, feature_list):
    # Set the parameters by cross-validation
    tuned_parameters = [{'max_depth': [5,10,20],
                         'n_estimators': [100, 200, 400]}]

    rf = RandomForestClassifier(random_state=0)

    for i in range(y.shape[1]):
        print('==================================================')
        print('Prediction of ' + lbl_colname[i])

        y_true = y[:,i]

        clf = GridSearchCV(rf, tuned_parameters, scoring='f1_macro', cv=5)
        clf.fit(X, y_true)

        print("Best parameters set found on development set:")
        print()
        print(clf.best_params_)
        print()
        print("Grid scores on development set:")
        print()
        means = clf.cv_results_['mean_test_score']
        stds = clf.cv_results_['std_test_score']
        for mean, std, params in zip(means, stds, clf.cv_results_['params']):
            print("%0.3f (+/-%0.03f) for %r"
                  % (mean, std * 2, params))
        print()

        y_pred = cross_val_predict(clf.best_estimator_, X, y_true, cv=5)

        print(f"count {sum(y_pred)}/{len(y_pred)}")
        print("Detailed classification report:")
        print()
        print(classification_report(y_true, y_pred))
        print()

        for i,v in sorted(zip(feature_list, clf.best_estimator_.feature_importances_), key=lambda tup: tup[1], reverse=True)[:20]:
            print('Feature: %s, Score: %.5f' % (i,v))

### All Features

In [12]:
X, y, feature_list = convert2np(data_df, lbl_colname)

In [13]:
hfpef_rf(X, y, feature_list)

Prediction of Death
Best parameters set found on development set:

{'max_depth': 10, 'n_estimators': 100}

Grid scores on development set:

0.542 (+/-0.205) for {'max_depth': 5, 'n_estimators': 100}
0.542 (+/-0.205) for {'max_depth': 5, 'n_estimators': 200}
0.542 (+/-0.205) for {'max_depth': 5, 'n_estimators': 400}
0.705 (+/-0.450) for {'max_depth': 10, 'n_estimators': 100}
0.705 (+/-0.450) for {'max_depth': 10, 'n_estimators': 200}
0.705 (+/-0.450) for {'max_depth': 10, 'n_estimators': 400}
0.705 (+/-0.450) for {'max_depth': 20, 'n_estimators': 100}
0.705 (+/-0.450) for {'max_depth': 20, 'n_estimators': 200}
0.705 (+/-0.450) for {'max_depth': 20, 'n_estimators': 400}

count 18/405
Detailed classification report:

              precision    recall  f1-score   support

           0       0.93      1.00      0.96       358
           1       1.00      0.38      0.55        47

    accuracy                           0.93       405
   macro avg       0.96      0.69      0.76       405
weig

### Selected Features

In [14]:
X, y, feature_list = convert2np(data_df, lbl_colname, selected_feat=True)

In [15]:
X.shape

(405, 13)

In [16]:
hfpef_rf(X, y, feature_list)

Prediction of Death
Best parameters set found on development set:

{'max_depth': 10, 'n_estimators': 100}

Grid scores on development set:

0.542 (+/-0.205) for {'max_depth': 5, 'n_estimators': 100}
0.559 (+/-0.226) for {'max_depth': 5, 'n_estimators': 200}
0.544 (+/-0.187) for {'max_depth': 5, 'n_estimators': 400}
0.705 (+/-0.450) for {'max_depth': 10, 'n_estimators': 100}
0.705 (+/-0.450) for {'max_depth': 10, 'n_estimators': 200}
0.705 (+/-0.450) for {'max_depth': 10, 'n_estimators': 400}
0.705 (+/-0.450) for {'max_depth': 20, 'n_estimators': 100}
0.705 (+/-0.450) for {'max_depth': 20, 'n_estimators': 200}
0.705 (+/-0.450) for {'max_depth': 20, 'n_estimators': 400}

count 18/405
Detailed classification report:

              precision    recall  f1-score   support

           0       0.93      1.00      0.96       358
           1       1.00      0.38      0.55        47

    accuracy                           0.93       405
   macro avg       0.96      0.69      0.76       405
weig

### Cleaned Data

In [17]:
X, y, feature_list = convert2np(data_df_nomiss, lbl_colname, selected_feat=True)

In [18]:
hfpef_rf(X, y, feature_list)

Prediction of Death
Best parameters set found on development set:

{'max_depth': 10, 'n_estimators': 100}

Grid scores on development set:

0.580 (+/-0.247) for {'max_depth': 5, 'n_estimators': 100}
0.580 (+/-0.247) for {'max_depth': 5, 'n_estimators': 200}
0.580 (+/-0.247) for {'max_depth': 5, 'n_estimators': 400}
0.687 (+/-0.511) for {'max_depth': 10, 'n_estimators': 100}
0.687 (+/-0.511) for {'max_depth': 10, 'n_estimators': 200}
0.687 (+/-0.511) for {'max_depth': 10, 'n_estimators': 400}
0.687 (+/-0.511) for {'max_depth': 20, 'n_estimators': 100}
0.687 (+/-0.511) for {'max_depth': 20, 'n_estimators': 200}
0.687 (+/-0.511) for {'max_depth': 20, 'n_estimators': 400}

count 6/145
Detailed classification report:

              precision    recall  f1-score   support

           0       0.95      1.00      0.97       132
           1       1.00      0.46      0.63        13

    accuracy                           0.95       145
   macro avg       0.97      0.73      0.80       145
weigh

C:\Users\sguly\miniconda3\envs\hfpef\lib\site-packages\sklearn\model_selection\_split.py:676: UserWarning: The least populated class in y has only 2 members, which is less than n_splits=5.
  warnings.warn(


Best parameters set found on development set:

{'max_depth': 5, 'n_estimators': 100}

Grid scores on development set:

0.796 (+/-0.498) for {'max_depth': 5, 'n_estimators': 100}
0.796 (+/-0.498) for {'max_depth': 5, 'n_estimators': 200}
0.796 (+/-0.498) for {'max_depth': 5, 'n_estimators': 400}
0.796 (+/-0.498) for {'max_depth': 10, 'n_estimators': 100}
0.796 (+/-0.498) for {'max_depth': 10, 'n_estimators': 200}
0.796 (+/-0.498) for {'max_depth': 10, 'n_estimators': 400}
0.796 (+/-0.498) for {'max_depth': 20, 'n_estimators': 100}
0.796 (+/-0.498) for {'max_depth': 20, 'n_estimators': 200}
0.796 (+/-0.498) for {'max_depth': 20, 'n_estimators': 400}



C:\Users\sguly\miniconda3\envs\hfpef\lib\site-packages\sklearn\model_selection\_split.py:676: UserWarning: The least populated class in y has only 2 members, which is less than n_splits=5.
  warnings.warn(


count 0/145
Detailed classification report:

              precision    recall  f1-score   support

           0       0.99      1.00      0.99       143
           1       0.00      0.00      0.00         2

    accuracy                           0.99       145
   macro avg       0.49      0.50      0.50       145
weighted avg       0.97      0.99      0.98       145


Feature: LA diameter, Score: 0.15891
Feature: Age, Score: 0.12586
Feature: Cr, Score: 0.12517
Feature: MAP, Score: 0.09705
Feature: medial a', Score: 0.08325
Feature: medial E', Score: 0.07605
Feature: NTProBNP, Score: 0.07274
Feature: LAVI, Score: 0.06460
Feature: CKD stage, Score: 0.06092
Feature: PP, Score: 0.06036
Feature: GFR, Score: 0.05586
Feature: Sex, Score: 0.01923
Feature: AF, Score: 0.00000
Prediction of Major cardiac events


C:\Users\sguly\miniconda3\envs\hfpef\lib\site-packages\sklearn\metrics\_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\sguly\miniconda3\envs\hfpef\lib\site-packages\sklearn\metrics\_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\sguly\miniconda3\envs\hfpef\lib\site-packages\sklearn\metrics\_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


Best parameters set found on development set:

{'max_depth': 10, 'n_estimators': 100}

Grid scores on development set:

0.807 (+/-0.199) for {'max_depth': 5, 'n_estimators': 100}
0.810 (+/-0.219) for {'max_depth': 5, 'n_estimators': 200}
0.808 (+/-0.222) for {'max_depth': 5, 'n_estimators': 400}
0.857 (+/-0.342) for {'max_depth': 10, 'n_estimators': 100}
0.844 (+/-0.361) for {'max_depth': 10, 'n_estimators': 200}
0.857 (+/-0.342) for {'max_depth': 10, 'n_estimators': 400}
0.850 (+/-0.351) for {'max_depth': 20, 'n_estimators': 100}
0.844 (+/-0.361) for {'max_depth': 20, 'n_estimators': 200}
0.850 (+/-0.351) for {'max_depth': 20, 'n_estimators': 400}

count 95/145
Detailed classification report:

              precision    recall  f1-score   support

           0       0.84      0.81      0.82        52
           1       0.89      0.91      0.90        93

    accuracy                           0.88       145
   macro avg       0.87      0.86      0.86       145
weighted avg       0.88 

##  Neural Networks

In [19]:
from sklearn.neural_network import MLPClassifier
from sklearn import preprocessing

In [20]:
def hfpef_ann(X, y, feature_list):
    # Set the parameters by cross-validation
    tuned_parameters = [{'hidden_layer_sizes': [(20,20,20), (100),(20,20)]}]

    ann = MLPClassifier(activation='tanh', max_iter=1000, random_state=0)

    for i in range(y.shape[1]):
        print('==================================================')
        print('Prediction of ' + lbl_colname[i])

        y_true = y[:,i]

        clf = GridSearchCV(ann, tuned_parameters, scoring='f1_macro', cv=5)
        clf.fit(X, y_true)

        print("Best parameters set found on development set:")
        print()
        print(clf.best_params_)
        print()
        print("Grid scores on development set:")
        print()
        means = clf.cv_results_['mean_test_score']
        stds = clf.cv_results_['std_test_score']
        for mean, std, params in zip(means, stds, clf.cv_results_['params']):
            print("%0.3f (+/-%0.03f) for %r"
                  % (mean, std * 2, params))
        print()

        y_pred = cross_val_predict(clf.best_estimator_, X, y_true, cv=5)

        print(f"count {sum(y_pred)}/{len(y_pred)}")
        print("Detailed classification report:")
        print()
        print(classification_report(y_true, y_pred))
        print()

### All Features

In [21]:
X, y, feature_list = convert2np(data_df, lbl_colname)

from sklearn.decomposition import PCA

X = PCA(n_components=0.95).fit_transform(X)
# scaler = preprocessing.MinMaxScaler().fit(X)
# X = scaler.transform(X)

In [22]:
X.shape

(405, 1)

In [23]:
hfpef_ann(X, y, feature_list)

Prediction of Death
Best parameters set found on development set:

{'hidden_layer_sizes': 100}

Grid scores on development set:

0.469 (+/-0.003) for {'hidden_layer_sizes': (20, 20, 20)}
0.539 (+/-0.149) for {'hidden_layer_sizes': 100}
0.469 (+/-0.003) for {'hidden_layer_sizes': (20, 20)}

count 99/405
Detailed classification report:

              precision    recall  f1-score   support

           0       0.91      0.77      0.83       358
           1       0.18      0.38      0.25        47

    accuracy                           0.73       405
   macro avg       0.54      0.58      0.54       405
weighted avg       0.82      0.73      0.77       405


Prediction of CV death
Best parameters set found on development set:

{'hidden_layer_sizes': (20, 20, 20)}

Grid scores on development set:

0.491 (+/-0.000) for {'hidden_layer_sizes': (20, 20, 20)}
0.483 (+/-0.104) for {'hidden_layer_sizes': 100}
0.491 (+/-0.000) for {'hidden_layer_sizes': (20, 20)}

count 0/405
Detailed classificat

C:\Users\sguly\miniconda3\envs\hfpef\lib\site-packages\sklearn\metrics\_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\sguly\miniconda3\envs\hfpef\lib\site-packages\sklearn\metrics\_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\sguly\miniconda3\envs\hfpef\lib\site-packages\sklearn\metrics\_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


Best parameters set found on development set:

{'hidden_layer_sizes': (20, 20, 20)}

Grid scores on development set:

0.604 (+/-0.026) for {'hidden_layer_sizes': (20, 20, 20)}
0.601 (+/-0.023) for {'hidden_layer_sizes': 100}
0.601 (+/-0.016) for {'hidden_layer_sizes': (20, 20)}

count 102/405
Detailed classification report:

              precision    recall  f1-score   support

           0       0.56      0.90      0.69       188
           1       0.81      0.38      0.52       217

    accuracy                           0.62       405
   macro avg       0.69      0.64      0.60       405
weighted avg       0.69      0.62      0.60       405




==================================================
Prediction of Death
Best parameters set found on development set:

{'hidden_layer_sizes': (20, 20, 20)}

Grid scores on development set:

0.519 (+/-0.125) for {'hidden_layer_sizes': (20, 20, 20)}
0.473 (+/-0.071) for {'hidden_layer_sizes': 100}
0.458 (+/-0.006) for {'hidden_layer_sizes': (20, 20)}

count 11/200
Detailed classification report:

              precision    recall  f1-score   support

           0       0.85      0.95      0.90       169
           1       0.27      0.10      0.14        31

    accuracy                           0.82       200
   macro avg       0.56      0.52      0.52       200
weighted avg       0.76      0.82      0.78       200


==================================================
Prediction of CV death
Best parameters set found on development set:

{'hidden_layer_sizes': 100}

Grid scores on development set:

0.485 (+/-0.007) for {'hidden_layer_sizes': (20, 20, 20)}
0.486 (+/-0.005) for {'hidden_layer_sizes': 100}
0.486 (+/-0.005) for {'hidden_layer_sizes': (20, 20)}

count 0/200
Detailed classification report:

              precision    recall  f1-score   support

           0       0.94      1.00      0.97       189
           1       0.00      0.00      0.00        11

    accuracy                           0.94       200
   macro avg       0.47      0.50      0.49       200
weighted avg       0.89      0.94      0.92       200


==================================================
Prediction of Major cardiac events

{'hidden_layer_sizes': (20, 20, 20)}

Grid scores on development set:

0.685 (+/-0.132) for {'hidden_layer_sizes': (20, 20, 20)}
0.661 (+/-0.076) for {'hidden_layer_sizes': 100}
0.666 (+/-0.103) for {'hidden_layer_sizes': (20, 20)}

count 111/200
Detailed classification report:

              precision    recall  f1-score   support

           0       0.71      0.64      0.67        99
           1       0.68      0.74      0.71       101

    accuracy                           0.69       200
   macro avg       0.69      0.69      0.69       200
weighted avg       0.69      0.69      0.69       200

### Selected Features

In [24]:
X, y, feature_list = convert2np(data_df, lbl_colname, selected_feat=True)

X = PCA(n_components=0.95).fit_transform(X)

# scaler = preprocessing.MinMaxScaler().fit(X)
# X = scaler.transform(X)

In [25]:
hfpef_ann(X, y, feature_list)

Prediction of Death
Best parameters set found on development set:

{'hidden_layer_sizes': 100}

Grid scores on development set:

0.469 (+/-0.003) for {'hidden_layer_sizes': (20, 20, 20)}
0.538 (+/-0.152) for {'hidden_layer_sizes': 100}
0.469 (+/-0.003) for {'hidden_layer_sizes': (20, 20)}

count 100/405
Detailed classification report:

              precision    recall  f1-score   support

           0       0.90      0.77      0.83       358
           1       0.18      0.38      0.24        47

    accuracy                           0.73       405
   macro avg       0.54      0.58      0.54       405
weighted avg       0.82      0.73      0.76       405


Prediction of CV death
Best parameters set found on development set:

{'hidden_layer_sizes': (20, 20, 20)}

Grid scores on development set:

0.491 (+/-0.000) for {'hidden_layer_sizes': (20, 20, 20)}
0.490 (+/-0.107) for {'hidden_layer_sizes': 100}
0.491 (+/-0.000) for {'hidden_layer_sizes': (20, 20)}

count 0/405
Detailed classifica

C:\Users\sguly\miniconda3\envs\hfpef\lib\site-packages\sklearn\metrics\_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\sguly\miniconda3\envs\hfpef\lib\site-packages\sklearn\metrics\_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\sguly\miniconda3\envs\hfpef\lib\site-packages\sklearn\metrics\_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


Best parameters set found on development set:

{'hidden_layer_sizes': (20, 20, 20)}

Grid scores on development set:

0.604 (+/-0.026) for {'hidden_layer_sizes': (20, 20, 20)}
0.601 (+/-0.023) for {'hidden_layer_sizes': 100}
0.601 (+/-0.016) for {'hidden_layer_sizes': (20, 20)}

count 102/405
Detailed classification report:

              precision    recall  f1-score   support

           0       0.56      0.90      0.69       188
           1       0.81      0.38      0.52       217

    accuracy                           0.62       405
   macro avg       0.69      0.64      0.60       405
weighted avg       0.69      0.62      0.60       405




### Cleaned Data

In [26]:
X, y, feature_list = convert2np(data_df_nomiss, lbl_colname, selected_feat=True)

# scaler = preprocessing.StandardScaler().fit(X)
# X = scaler.transform(X)

In [27]:
hfpef_ann(X, y, feature_list)

Prediction of Death
Best parameters set found on development set:

{'hidden_layer_sizes': (20, 20, 20)}

Grid scores on development set:

0.476 (+/-0.009) for {'hidden_layer_sizes': (20, 20, 20)}
0.476 (+/-0.009) for {'hidden_layer_sizes': 100}
0.476 (+/-0.009) for {'hidden_layer_sizes': (20, 20)}

count 0/145
Detailed classification report:



C:\Users\sguly\miniconda3\envs\hfpef\lib\site-packages\sklearn\metrics\_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\sguly\miniconda3\envs\hfpef\lib\site-packages\sklearn\metrics\_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\sguly\miniconda3\envs\hfpef\lib\site-packages\sklearn\metrics\_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\sguly\mini

              precision    recall  f1-score   support

           0       0.91      1.00      0.95       132
           1       0.00      0.00      0.00        13

    accuracy                           0.91       145
   macro avg       0.46      0.50      0.48       145
weighted avg       0.83      0.91      0.87       145


Prediction of CV death
Best parameters set found on development set:

{'hidden_layer_sizes': (20, 20, 20)}

Grid scores on development set:

0.796 (+/-0.498) for {'hidden_layer_sizes': (20, 20, 20)}
0.796 (+/-0.498) for {'hidden_layer_sizes': 100}
0.796 (+/-0.498) for {'hidden_layer_sizes': (20, 20)}



C:\Users\sguly\miniconda3\envs\hfpef\lib\site-packages\sklearn\model_selection\_split.py:676: UserWarning: The least populated class in y has only 2 members, which is less than n_splits=5.
  warnings.warn(


count 0/145
Detailed classification report:

              precision    recall  f1-score   support

           0       0.99      1.00      0.99       143
           1       0.00      0.00      0.00         2

    accuracy                           0.99       145
   macro avg       0.49      0.50      0.50       145
weighted avg       0.97      0.99      0.98       145


Prediction of Major cardiac events


C:\Users\sguly\miniconda3\envs\hfpef\lib\site-packages\sklearn\metrics\_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\sguly\miniconda3\envs\hfpef\lib\site-packages\sklearn\metrics\_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\sguly\miniconda3\envs\hfpef\lib\site-packages\sklearn\metrics\_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


Best parameters set found on development set:

{'hidden_layer_sizes': 100}

Grid scores on development set:

0.429 (+/-0.093) for {'hidden_layer_sizes': (20, 20, 20)}
0.645 (+/-0.267) for {'hidden_layer_sizes': 100}
0.391 (+/-0.013) for {'hidden_layer_sizes': (20, 20)}

count 123/145
Detailed classification report:

              precision    recall  f1-score   support

           0       0.82      0.35      0.49        52
           1       0.72      0.96      0.82        93

    accuracy                           0.74       145
   macro avg       0.77      0.65      0.66       145
weighted avg       0.76      0.74      0.70       145


